File: src-app/src/lib/adminConfig.ts
/**
 * Shared allowlist for internal admin dashboards.
 * Only these email accounts may access the Marketing Growth OS
 * and Analytics Dashboard — in the sidebar, via routes, and via API.
 */
export const ALLOWED_INTERNAL_DASHBOARD_EMAILS: string[] = [
  'queensworld1984@gmail.com',
  'dorothypierce84@gmail.com',
]

/** Returns true if the given email is in the internal dashboard allowlist. */
export function isInternalDashboardEmail(email: string | null | undefined): boolean {
  if (!email) return false
  const lower = email.toLowerCase()
  return ALLOWED_INTERNAL_DASHBOARD_EMAILS.some(e => e.toLowerCase() === lower)
}

File: backend/app/utils/admin_config.py
"""
Shared admin access configuration for internal dashboards.

Only accounts listed in ALLOWED_INTERNAL_DASHBOARD_EMAILS may access:
  - Marketing Growth OS  (/api/growth/* — non-public endpoints)
  - Analytics Dashboard  (/api/admin/analytics/*)

Use require_internal_dashboard as a FastAPI Depends() in any route that
must be restricted to these accounts.  The same list is mirrored on the
frontend in src/lib/adminConfig.ts — keep both in sync.
"""
from fastapi import Depends, HTTPException
from app.utils.auth import get_current_user

ALLOWED_INTERNAL_DASHBOARD_EMAILS: list[str] = [
    "queensworld1984@gmail.com",
    "dorothypierce84@gmail.com",
]

_ALLOWED_LOWER: list[str] = [e.lower() for e in ALLOWED_INTERNAL_DASHBOARD_EMAILS]


def require_internal_dashboard(user: dict = Depends(get_current_user)) -> dict:
    """FastAPI dependency — restrict access to internal dashboard allowlist."""
    if user.get("email", "").lower() not in _ALLOWED_LOWER:
        raise HTTPException(status_code=403, detail="Unauthorized")
    return user
